api/sda/sda.go

```diff
@@ -206,90 +196,104 @@ func Download(c *gin.Context) {
+	qEnd := c.DefaultQuery("endCoordinate", "0")
 
-// stitchFile stitches the header and file body together for Crypt4GHReader
-// and returns a streamable Reader
-var stitchFile = func(header []byte, file io.ReadCloser, coordinates *headers.DataEditListHeaderPacket) (*streaming.Crypt4GHReader, error) {
-	log.Debugf("stitching header to file %s for streaming", file)
-	// Stitch header and file body together
-	hr := bytes.NewReader(header)
-	mr := io.MultiReader(hr, file)
-	c4ghr, err := streaming.NewCrypt4GHReader(mr, *config.Config.App.Crypt4GHKey, coordinates)
+	// Parse and verify coordinates are valid
+	start, err := strconv.ParseInt(qStart, 10, 0)
 	if err != nil {
-		log.Errorf("failed to create Crypt4GH stream reader, %v", err)
+		log.Errorf("failed to convert start coordinate %d to integer, %s", start, err)
+		c.String(http.StatusInternalServerError, "startCoordinate must be an integer")
 
-		return nil, err
+		return
 	}
-	log.Debugf("file stream for %s constructed", file)
+	end, err := strconv.ParseInt(qEnd, 10, 0)
+	if err != nil {
+		log.Errorf("failed to convert end coordinate %d to integer, %s", end, err)
 
-	return c4ghr, nil
-}
+		c.String(http.StatusInternalServerError, "endCoordinate must be an integer")
 
-// parseCoordinates takes query param coordinates and converts them to
-// Crypt4GH reader format
-var parseCoordinates = func(r *http.Request) (*headers.DataEditListHeaderPacket, error) {
+		return
+	}
+	if end < start {
+		log.Errorf("endCoordinate=%d must be greater than startCoordinate=%d", end, start)
 
-	coordinates := &headers.DataEditListHeaderPacket{}
+		c.String(http.StatusInternalServerError, "endCoordinate must be greater than startCoordinate")
 
-	// Get query params
-	qStart := r.URL.Query().Get("startCoordinate")
-	qEnd := r.URL.Query().Get("endCoordinate")
+		return
+	}
 
-	// Parse and verify coordinates are valid
-	if len(qStart) > 0 && len(qEnd) > 0 {
-		start, err := strconv.ParseUint(qStart, 10, 64)
-		if err != nil {
-			log.Errorf("failed to convert start coordinate %d to integer, %s", start, err)
+	sendStream(c.Writer, c4ghr, start, end)
 
-			return nil, errors.New("startCoordinate must be an integer")
-		}
-		end, err := strconv.ParseUint(qEnd, 10, 64)
-		if err != nil {
-			log.Errorf("failed to convert end coordinate %d to integer, %s", end, err)
+}
 
-			return nil, errors.New("endCoordinate must be an integer")
-		}
-		if end < start {
-			log.Errorf("endCoordinate=%d must be greater than startCoordinate=%d", end, start)
+// sendStream streams file contents from a reader
+var sendStream = func(w http.ResponseWriter, c4ghr *streaming.Crypt4GHReader, start, end int64) {
+	log.Debug("begin data stream")
+
+	if start != 0 {
+		// We don't want to read from start, skip ahead to where we should be
+		if _, err := c4ghr.Seek(start, 0); err != nil {
+			w.WriteHeader(500)
+			_, err := w.Write([]byte(fmt.Sprintf("error occurred %v", err)))
+			if err != nil {
+				return
+			}
```

Comment:
```suggestion
			w.Write([]byte(fmt.Sprintf("error occurred %v", err)))
```

We're doing a valueless return afterwards anyway (within the seek failed branch).

(Not tested, could possibly have linter opinions causing an explicit ignore to be better.)

Reply:
`api/sda/sda.go:250:11: Error return value of `w.Write` is not checked (errcheck)` 

i will return an error there